pkg/microservice/aslan/core/environment/service/environment.go

```diff
@@ -1689,6 +1705,41 @@ func GetHelmChartVersions(productName, envName string, log *zap.SugaredLogger) (
 	return helmVersions, nil
 }
 
+func DeleteProductServices(envName, productName string, serviceNames []string, log *zap.SugaredLogger) (err error) {
+	productInfo, err := commonrepo.NewProductColl().Find(&commonrepo.ProductFindOptions{Name: productName, EnvName: envName})
+	if err != nil {
+		log.Errorf("find product error: %v", err)
+		return err
+	}
+
+	for serviceGroupIndex, serviceGroup := range productInfo.Services {
```

Comment:
Would it be better if we drop the service first and then update the database instead?